v4/export/sql_test.go

```diff
@@ -1178,6 +1187,326 @@ func (s *testSQLSuite) TestBuildRegionQueriesWithPartitions(c *C) {
 	}
 }
 
+func buildMockNewRows(mock sqlmock.Sqlmock, columns []string, driverValues [][]driver.Value) *sqlmock.Rows {
+	rows := mock.NewRows(columns)
+	for _, driverValue := range driverValues {
+		rows.AddRow(driverValue...)
+	}
+	return rows
+}
+
+func readRegionCsvDriverValues(c *C) [][]driver.Value {
+	// nolint: dogsled
+	_, filename, _, _ := runtime.Caller(0)
+	csvFilename := path.Join(path.Dir(filename), "region_results.csv")
+	file, err := os.Open(csvFilename)
+	c.Assert(err, IsNil)
+	csvReader := csv.NewReader(file)
+	values := make([][]driver.Value, 0, 990)
+	for {
+		results, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		c.Assert(err, IsNil)
+		if len(results) != 3 {
+			continue
+		}
+		regionID, err := strconv.Atoi(results[0])
+		c.Assert(err, IsNil)
+		startKey, endKey := results[1], results[2]
+		values = append(values, []driver.Value{regionID, startKey, endKey})
+	}
+	return values
+}
+
+func (s *testSQLSuite) TestBuildVersion3RegionQueries(c *C) {
+	db, mock, err := sqlmock.New()
+	c.Assert(err, IsNil)
+	defer db.Close()
+	conn, err := db.Conn(context.Background())
+	c.Assert(err, IsNil)
+	tctx, cancel := tcontext.Background().WithLogger(appLogger).WithCancel()
+	oldOpenFunc := openDBFunc
+	defer func() {
+		openDBFunc = oldOpenFunc
+	}()
+	openDBFunc = func(_, _ string) (*sql.DB, error) {
+		return db, nil
+	}
+
+	conf := DefaultConfig()
+	conf.ServerInfo = ServerInfo{
+		HasTiKV:       true,
+		ServerType:    ServerTypeTiDB,
+		ServerVersion: decodeRegionVersion,
+	}
+	database := "test"
+	conf.Tables = DatabaseTables{
+		database: []*TableInfo{
+			{"t1", TableTypeBase},
+			{"t2", TableTypeBase},
+			{"t3", TableTypeBase},
+			{"t4", TableTypeBase},
+		},
+	}
+	d := &Dumper{
+		tctx:                      tctx,
+		conf:                      conf,
+		cancelCtx:                 cancel,
+		selectTiDBTableRegionFunc: selectTiDBTableRegion,
+	}
+	showStatsHistograms := buildMockNewRows(mock, []string{"Db_name", "Table_name", "Partition_name", "Column_name", "Is_index", "Update_time", "Distinct_count", "Null_count", "Avg_col_size", "Correlation"},
+		[][]driver.Value{
+			{"test", "t2", "p0", "a", 0, "2021-06-27 17:43:51", 1999999, 0, 8, 0},
```

Comment:
`a` is a non-index column, select by `a` with WHERE will cause multiple TableScan?

Reply:
Don't understand. The result of `show stats_histograms` will only be used to get partition IDs. It will not be used to split tables.